airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence2.java

```diff
@@ -0,0 +1,1911 @@
+                .orElseThrow())
+        .withNamespaceFormat(record.get(CONNECTION.NAMESPACE_FORMAT))
+        .withPrefix(record.get(CONNECTION.PREFIX))
+        .withSourceId(record.get(CONNECTION.SOURCE_ID))
+        .withDestinationId(record.get(CONNECTION.DESTINATION_ID))
+        .withName(record.get(CONNECTION.NAME))
+        .withCatalog(Jsons.deserialize(record.get(CONNECTION.CATALOG).data(), ConfiguredAirbyteCatalog.class))
+        .withStatus(record.get(CONNECTION.STATUS) == null ? null
+            : Enums.toEnum(record.get(CONNECTION.STATUS, String.class), StandardSync.Status.class).orElseThrow())
+        .withSchedule(Jsons.deserialize(record.get(CONNECTION.SCHEDULE).data(), Schedule.class))
+        .withManual(record.get(CONNECTION.MANUAL))
+        .withOperationIds(connectionOperationIds(record.get(CONNECTION.ID)))
+        .withResourceRequirements(Jsons.deserialize(record.get(CONNECTION.RESOURCE_REQUIREMENTS).data(), ResourceRequirements.class));
+  }
+
+  private StandardSyncState getStandardSyncState(final String configId) throws IOException, ConfigNotFoundException {
+    final Result<Record> result = database.query(ctx -> ctx.select(asterisk())
+        .from(STATE)
+        .where(STATE.CONNECTION_ID.eq(UUID.fromString(configId)))
+        .fetch());
+
+    validate(configId, result, ConfigSchema.STANDARD_SYNC_STATE);
+    final Record record = result.get(0);
+
+    return new StandardSyncState()
+        .withConnectionId(record.get(STATE.CONNECTION_ID))
+        .withState(Jsons.deserialize(record.get(STATE.STATE_).data(), State.class));
+  }
+
+  private List<UUID> connectionOperationIds(final UUID connectionId) throws IOException {
+    final Result<Record> result = database.query(ctx -> ctx.select(asterisk())
+        .from(CONNECTION_OPERATION)
+        .where(CONNECTION_OPERATION.CONNECTION_ID.eq(connectionId))
+        .fetch());
+
+    final List<UUID> ids = new ArrayList<>();
+    for (Record record : result) {
+      ids.add(record.get(CONNECTION_OPERATION.OPERATION_ID));
+    }
+
+    return ids;
+  }
+
+  private void validate(final String configId, final Result<Record> result, final AirbyteConfig airbyteConfig) throws ConfigNotFoundException {
+    if (result.isEmpty()) {
+      throw new ConfigNotFoundException(airbyteConfig, configId);
+    } else if (result.size() > 1) {
+      throw new IllegalStateException(String.format("Multiple %s configs found for ID %s: %s", airbyteConfig, configId, result));
+    }
+  }
+
+  @Override
+  public <T> List<T> listConfigs(AirbyteConfig configType, Class<T> clazz) throws JsonValidationException, IOException {
+    if (!useNewPersistence()) {
+      return oldConfigPersistence.listConfigs(configType, clazz);
+    }
+    final List<T> config = new ArrayList<>();
+    if (configType == ConfigSchema.STANDARD_WORKSPACE) {
+      getStandardWorkspaceWithMetadata().forEach(c -> config.add((T) c.getConfig()));
```

Comment:
should all these methods start with `list` instead of `get`? my initial reaction was that these were retrieving one record instead of all the records like it's doing.